### PR TITLE
Get-WmiObject - Invalid class "Win32_Service" - Workaround

### PR DESCRIPTION
**Pull request by stryngs**
## What does this PR do?
This PR adds a workaround to a situation in Windows where calling Get-WmiObject returns "Win32_Service" as an Invalid class.

## Why is it important?
This is important to users which remotely deploy and install any of the Beats in a Windows environment.  Rather than solving the class issue I focused on a workaround which can be provided until a proper solution to the class issue is found.

## Checklist

- [ ] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Author's Checklist

- [ ] N/A

## How to test this PR locally
Build any Beats package and then run the associated installer for Windows.

## Related issues
- N/A

## Use cases
Windows zipfile deployments for service installations.

## Screenshots
N/A

## Logs
N/A

### CODE REVIEW

**Comment by stryngs** [comment]
> x Author of the following commits did not sign a [Contributor Agreement](https://www.elastic.co/contributor-agreement): [257f588](https://github.com/elastic/beats/commit/257f5883a79ad9df46fdb9d6d010e0ed60ae785f), [a5eae20](https://github.com/elastic/beats/commit/a5eae20eb9aa9fa19f1e7ebe5dd0ca4d820df8e7)
> 
> Please, read and sign the above mentioned agreement if you want to contribute to this project

Signed.

**Comment by stryngs** [comment]
What do I need to do to keep this moving forward?

**Comment by stryngs** [comment]
Awesome and glad to hear it.

File: dev-tools/packaging/templates/windows/install-service.ps1.tmpl [hunk 1]
```diff
@@ -1,9 +1,18 @@
 # Delete and stop the service if it already exists.
 if (Get-Service {{.BeatName}} -ErrorAction SilentlyContinue) {
-  $service = Get-WmiObject -Class Win32_Service -Filter "name='{{.BeatName}}'"
-  $service.StopService()
-  Start-Sleep -s 1
-  $service.delete()
+  try {
```
Comment (leehinman) [inline dev-tools/packaging/templates/windows/install-service.ps1.tmpl:3]: Lets not do the try/catch and just change to the code in the catch block.

I tested the code in the catch block on Windows 11, Windows Server 2012, 2016 & 2019.  Works fine.

**Review by leehinman — CHANGES_REQUESTED** [review]
Will also need a changelog entry.

File: dev-tools/packaging/templates/windows/install-service.ps1.tmpl (hunk 1 above)
Comment (stryngs) [reply dev-tools/packaging/templates/windows/install-service.ps1.tmpl]: I made the requested changes, please let me know if there is anything further needed.  Thanks for moving so quick on this!

**Comment by stryngs** [comment]
What else needs to be done for this?

**Review by leehinman — APPROVED** [review]
LGTM

**Comment by cmacknz** [comment]
We just need the merge conflict in the changelog resolved and we can merge this.

**Comment by stryngs** [comment]
I'm not the best with git.  I tried the steps and got this:
```
~/beats$ git fetch upstream
fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

```

What can I do to fix?

**Comment by cmacknz** [comment]
Merged! This will be in the 8.6 release for Beats.

**Comment by stryngs** [comment]
I'm looking at 8.6.2 and still do not see this pulled in?

**Comment by cmacknz** [comment]
It should be in 8.6.2, Git certainly thinks it is looking at the tags on the merged commit https://github.com/elastic/beats/commit/4e58e317cb1158235f89a18d3e236c40ce323d7c

**Comment by stryngs** [comment]
We have 8.6.2 on site and I need to apply this change by hand.  Maybe someone didn't push a specific button or?

**Comment by cmacknz** [comment]
The commit is part of the 8.6.2 release, Git shows that it is and you can see it if you go far enough back in the 8.6.2 commit history. It is one of the last commits in https://github.com/elastic/beats/commits/v8.6.2?after=9b77c2c135c228c2eedc310f6e975bb1a76169b1+139&branch=v8.6.2&qualified_name=refs%2Ftags%2Fv8.6.2

If it isn't working it is possible something was missed as part of the fix here, possibly something that is part of the build and packaging step.

**Comment by BrunoMacias** [comment]
Was there more information about the root cause of win32_service issues?

**Comment by stryngs** [comment]
> The commit is part of the 8.6.2 release, Git shows that it is and you can see it if you go far enough back in the 8.6.2 commit history. It is one of the last commits in https://github.com/elastic/beats/commits/v8.6.2?after=9b77c2c135c228c2eedc310f6e975bb1a76169b1+139&branch=v8.6.2&qualified_name=refs%2Ftags%2Fv8.6.2
> 
> If it isn't working it is possible something was missed as part of the fix here, possibly something that is part of the build and packaging step.

This never made it into the release.  I am currently working at 8.11.x and not seeing it.

**Pull request merged**